### DISCUSSION TITLE
Enabling the robots.txt

### DISCUSSION THREAD

**Discussion by RealPresidentlin** [plausible/analytics#1798] [Self-Hosted Support] [answered]
Hi 

For some reason, Google is indexing a number of the /api/event

I was wondering if anyone has figured out a way to prevent this. 
My current aim is to enable the robots.txt and add a disallow but I am struggling to find it? 

Any suggestions would be welcomed.

**Comment by jvandenaardweg** [comment, marked as answer]
Yes, got this working on my self hosted instance, but not with a robots.txt file. If you use a reverse proxy like nginx, add the noindex nofollow header to all paths/locations:

```
add_header X-Robots-Tag "noindex, nofollow" always;
```

Then crawlers should ignore it. 

But once in the search index, it can only be deleted if you return a 404 for that url and request Google to delete it.

**Reply by lightson** [reply]
If you use google search console @Lwachira there's a 'removals' option that will have google delist the content in the meantime.

**Comment by MarcoGlauser** [comment]
The robots.txt seems to be located under /app/lib/plausible-0.0.1/priv/static/robots.txt in the docker image.
We chose to mount a local robots.txt as a docker volume instead and it works for now. Not sure if that path might change in the future.
```yaml
services:
  ...
  plausible:
    ...
    volumes:
      - ./robots.txt:/app/lib/plausible-0.0.1/priv/static/robots.txt
```

**Reply by paskal** [reply]
@ukutaht, does it make sense to make it equal to `User-agent: *\nDisallow: /\n` by default? It was an unpleasant surprise to me to learn that in addition to my one-hundred-page website, two thousand pages with different parameters from self-hosted Plausible instances were indexed by Google. If you want to keep it non-blocking, most URL parameters should be marked as denied so that doubles would not be indexed: please let me know which way you would prefer, and I will help you craft the right `robots.txt`.

**Reply by ruslandoga** [reply]
👋 @paskal 

Just FYI: I started working on it in https://github.com/plausible/analytics/pull/3905

**Reply by RealPresidentlin** [reply]
@ruslandoga Thank you for closing the loop on this, this will help a lot of people who aren't even aware. We are 5 days away from it being the 2 year anniversary of this discussion 😂

**Comment by madanyang** [comment]
I use a different approach with nginx configuration and add this to the server configuration 

```
location = /robots.txt {
add_header  Content-Type  text/plain;
return 200 "User-agent: *\nDisallow: /\n";
}
```

**Reply by sgbx** [reply]
Works. Thanks for the solution.
